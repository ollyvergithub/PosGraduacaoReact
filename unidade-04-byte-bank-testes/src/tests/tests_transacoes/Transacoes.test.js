import React from "react";
import {render} from "@testing-library/react";
import Transacoes from "../../transacoes/Transacoes";

describe('Componente Transações do extrato', () =>{

    it('Deve retornar sempre o mesmo snapshot', function () {

        const transacoes = [
            {
                data: "10/08/2020",
                id: 1,
                transacao: "saque",
                valor: '1',
            },
            {
                data: "10/08/2020",
                id: 2,
                transacao: "deposito",
                valor: '2',
            },
        ]

        const {container} = render(
            <Transacoes
                transacoes={transacoes}
            />
        )

        expect(container.firstChild).toMatchSnapshot()

    });
})
