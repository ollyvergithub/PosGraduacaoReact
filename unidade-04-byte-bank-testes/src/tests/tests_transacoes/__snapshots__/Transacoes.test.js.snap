// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Transações do extrato Deve retornar sempre o mesmo snapshot 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      10/08/2020
    </p>
    <p>
      saque
    </p>
    <p>
      R$ 
      1
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      10/08/2020
    </p>
    <p>
      deposito
    </p>
    <p>
      R$ 
      2
    </p>
  </div>
</div>
`;
